### PR TITLE
[SDK-76] Feature/account linking update

### PR DESCRIPTION
**Pull request by HarrisonHough**
## Description

- Adding support for account linking events in WebGL example (template)
- Also a refactor of how the events are handled 

#### Added

-  support for user events 

#### Updated

-  refactor of classes and handling of web events

## How to Test

-   Add steps to locally test these changes

## Checklist

-   [ ] Documentation is updated.

## Details

-   If more details are necessary to support the description with images and videos add them here.

[SDK-76]: https://ready-player-me.atlassian.net/browse/SDK-76?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: Assets/Ready Player Me/Resources/Settings/CoreSettings.asset [hunk 1]
```diff
@@ -12,5 +12,5 @@ MonoBehaviour:
   m_Script: {fileID: 11500000, guid: f12696b20f2e49519c1b2a403a76778b, type: 3}
   m_Name: CoreSettings
   m_EditorClassIdentifier: 
-  Subdomain: demo
+  Subdomain: dev-sdk
```
Comment (gadamyan) [inline Assets/Ready Player Me/Resources/Settings/CoreSettings.asset:15]: Isn't this repo public?

File: Packages/manifest.json [hunk 2]
```diff
@@ -3,7 +3,7 @@
     "com.atteneder.gltfast": "https://github.com/atteneder/glTFast.git#v5.0.0",
     "com.readyplayerme.avatarloader": "https://github.com/readyplayerme/rpm-unity-sdk-avatar-loader.git",
     "com.readyplayerme.core": "https://github.com/readyplayerme/Unity-core.git",
-    "com.readyplayerme.webview": "https://github.com/readyplayerme/rpm-unity-sdk-webview.git",
+    "com.readyplayerme.webview": "https://github.com/readyplayerme/rpm-unity-sdk-webview.git#feature/account-linking-request",
```
Comment (gadamyan) [inline Packages/manifest.json:6]: Will you revert this line and the one in the next file before merging?

File: Assets/Scripts/WebInterface.cs [hunk 3]
```diff
@@ -1,34 +1,49 @@
...
-    public static void SetIFrameVisibility(bool isVisible)
-    {
-
+    
+    [DllImport("__Internal")]
+    private static extern void ReloadUrl(string url);
+#endif
+        public static void SetIFrameVisibility(bool isVisible)
+        {
+#if !UNITY_EDITOR && UNITY_WEBGL
```
Comment (gadamyan) [inline Assets/Scripts/WebInterface.cs:24]: Adding a lot of macros in a single class makes the example unreadable, Is there a way to move these macros to the upper level, or use interfaces to make the file more understandable

**Comment by gadamyan** [comment]
Could you update the Readme in this PR as well, Please also mention that they should not hardcode the ApiKey in the project, and make a request with  it.

File: Assets/Ready Player Me/Resources/Settings/CoreSettings.asset (hunk 1 above)
Comment (HarrisonHough) [reply Assets/Ready Player Me/Resources/Settings/CoreSettings.asset:15]: yep it is, will need to revert before merging, but thanks for reminding.
I just needed this subdomain to be able to test :D

File: Assets/Scripts/WebFrameHandler.cs [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+        
+        private void Start()
+        {
+            if (autoInitialize == AutoInitialize.OnStart)
+            {
+                Setup();
+            }
+        }
+
+        public void Setup(string loginToken = "")
```
[the remark this answers is not in the document]
Comment (HarrisonHough) [reply Assets/Scripts/WebFrameHandler.cs:40]: Yep I tried to make it as customizable as possible (perhaps too much). 
But if they want to instead run it with a token they can set AutoInitizialize to none and call the function from their own script with the token.

**Comment by HarrisonHough** [comment]
> Could you update the Readme in this PR as well, Please also mention that they should not hardcode the ApiKey in the project, and make a request with it.

yep good idea will do!

File: Assets/Scripts/WebInterface.cs (hunk 3 above)
Comment (HarrisonHough) [reply Assets/Scripts/WebInterface.cs:24]: yep it's tricky, I tried to isolate these macros to this script, previously I had 1 maco around this whole script. But then I also needed the macro's in the scripts that referenced these functions. 
But by moving these macro's inside these functions, other scripts can still reference them (they just won't do anything if the confiditons are not met). 
Perhaps @rYuuk you might have an idea on how to better handle it? 
I agree it makes it difficult to read though

**Review by gadamyan — APPROVED** [review]

**Review by rYuuk — APPROVED** [review]

**Pull request merged**